From 7c99c5454c7718a5acfc4b5056b54f2d3d1cb349 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 11 Nov 2024 21:02:00 +0100
Subject: [PATCH] window: Close popup on button click

Without this, the export popup remains open when the user clicks on a
button, and have to dismiss it before being able to use the export dialog.
---
 src/widgets/window.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/widgets/window.rs b/src/widgets/window.rs
index 715c719..6a40537 100644
--- a/src/widgets/window.rs
+++ b/src/widgets/window.rs
@@ -75,6 +75,7 @@ mod imp {
                 Some(glib::VariantTy::STRING),
                 |window, _, target| async move {
                     if let Some(project) = window.imp().open_project.borrow().as_ref() {
+                        window.imp().exporter.popdown();
                         let project_type = target.unwrap().get::<String>().unwrap();
                         let icon = crate::common::Icon::from(project_type);
                         if project.export(icon, &window).await.is_err() {
-- 
GitLab

